src/client/components/MealDetails.jsx

```diff
@@ -0,0 +1,162 @@
+import React, { useEffect, useState } from "react";
+import { Redirect, useParams, useHistory } from "react-router-dom";
+import "./styles.css";
+import MealCard from "./MealCard";
+
+const dtFormat = new Intl.DateTimeFormat("default", {
+  day: "2-digit",
+  month: "short",
+  year: "numeric",
+  hour: "numeric",
+  minute: "numeric",
+  hour12: false,
+});
+
+function MealDetails() {
+  const history = useHistory();
+  const { id } = useParams();
+  const [reservedMeal, setReservedMeal] = useState();
+  const [isResLoading, setIsResLoading] = useState(true);
+  const [fetchResError, setFetchResError] = useState(null);
+
+  // Reservation handling
+  const [resName, setResName] = useState("");
```

Comment:
The form logic (states and handlers) and the form UI, could be moved to a separate component for clarity.